openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1157 @@
+        self.bracketed_method = bracketed_method
+
+        check_type('tol', tol, Real)
+        self.tol = tol
+
+        check_type('target', target, Real)
+        self.target = target
+
+        self.print_iterations = print_iterations
+        self.search_for_keff_output = search_for_keff_output
+
+    @property
+    def bracketed_method(self):
+        return self._bracketed_method
+
+    @bracketed_method.setter
+    def bracketed_method(self, value):
+        check_value('bracketed_method', value, _SCALAR_BRACKETED_METHODS)
+        if value != 'brentq':
+            warn('WARNING: brentq bracketed method is recommended')
+        self._bracketed_method = value
+
+    @property
+    def tol(self):
+        return self._tol
+
+    @tol.setter
+    def tol(self, value):
+        check_type("tol", value, Real)
+        self._tol = value
+
+    @property
+    def target(self):
+        return self._target
+
+    @target.setter
+    def target(self, value):
+        check_type("target", value, Real)
+        self._target = value
+
+    @classmethod
+    def from_params(cls, obj, attr, operator, model, **kwargs):
+        return cls(obj, attr, operator, model, **kwargs)
+
+    @abstractmethod
+    def _model_builder(self, param):
+        """
+        Builds the parametric model to be passed to the
+        :meth:`openmc.search.search_for_keff` method.
+        Callable function which builds a model according to a passed
+        parameter. This function must return an openmc.model.Model object.
+        Parameters
+        ----------
+        param : parameter
+            model function variable
+        Returns
+        -------
+        _model :  openmc.model.Model
+            OpenMC parametric model
+        """
+
+    def _search_for_keff(self, val):
+        """
+        Perform the criticality search for a given parametric model.
+        If the solution lies off the initial bracket, this method iteratively
+        adapt it until :meth:`openmc.search.search_for_keff` return a valid
+        solution.
+        The adapting bracket algorithm takes the ratio between the bracket and
+        the returned keffs values to scale the bracket until a valid solution
+        is found.
+        A bracket limit pose the upper and lower boundaries to the adapting
+        bracket. If one limit is hit, the algorithm will stop and the closest
+        limit value will be used.
+        Parameters
+        ----------
+        val : float
+            Previous result value
+        Returns
+        -------
+        root : float
+            Estimated value of the variable parameter where keff is the
+            targeted value
+        """
+        #make sure we don't modify original bracket and tol values
+        bracket = deepcopy(self.bracket)
+
+        #search_for_keff tolerance should vary according to the first guess value
+        if abs(val) > 1.0:
+            tol = self.tol / abs(val)
+        else:
+            tol = self.tol
```

Comment:
Why does tolerance need to vary according to the first value? Is it supposed to be relative but it's given as absolute? Or vice versa. Or is `val` not a real number the first iteration?

Please consider removing this conversion